### PR TITLE
Sort data in connections

### LINKED ISSUE #8506: [UI] Connections: External Link and Light Mode Issue
<@U04N2G30PD0> <@U04TGEMEFDW> we have an issue here with poor contrast on the Connections page.

Also, <@U04TGEMEFDW> will you make the external link a &lt;sup&gt; (or look like a &lt;sup&gt;) of the Element name and make the element name clickable as well (with an underline on hover)?

[Slack Message](https://layer5io.slack.com/archives/C023QGMSH8D/p1692283242181489?thread_ts=1692283117.482789&cid=C023QGMSH8D)

### LINKED ISSUE #8511: Connection table automatically starts paginating into an infinite loop
<@U04TGEMEFDW> The connections table got stuck into an infinte loop automatically changes pages back and forth ,  the bug got triggered by clicking the next page quickly

[Slack Message](https://layer5io.slack.com/archives/C023QGMSH8D/p1692294078174919?thread_ts=1692294078.174919&cid=C023QGMSH8D)

### PR DESCRIPTION
**Pull request by senali-d**
**Notes for Reviewers**

This PR fixes #8511
This PR fixes #8506 

- [x] Sort connections datatable data
- [x]  Connection status update #8711 
- [x] #8511
- [x] #8506 
- [x] Connections bulk delete
- [x] Connections metadata show in when expand row

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/connections/index.js [hunk 1]
```diff
@@ -560,13 +611,14 @@ function Connections({
    * fetch connections when the page loads
    */
   useEffect(() => {
-    getConnections(page, pageSize, search);
+    getConnections(page, pageSize, search, sortOrder);
```
Comment (aabidsofi19) [inline ui/components/connections/index.js:614]: should'nt sortOrder also be in dependencies

Comment (aabidsofi19) [reply ui/components/connections/index.js:614]: if i am right changes are handled in onTableChangeAlso  also .

File: ui/components/connections/index.js [hunk 2]
```diff
@@ -313,10 +320,14 @@ function Connections({
...
+        sort : true,
+        sortThirdClickReset : true,
+        customHeadRender : function CustomHead({ index, ...column }, sortColumn, columnMeta) {
           return (
-            <TableCell key={index}>
-              <b>{column.label}</b>
+            <TableCell key={index} onClick={() => sortColumn(index)}>
+              <TableSortLabel active={columnMeta.name === column.name} direction={columnMeta.direction || "asc"}>
+                <b>{column.label}</b>
+              </TableSortLabel>
```
Comment (aabidsofi19) [inline ui/components/connections/index.js:330]: you can create a small reusable component `SortableTableCell` at top of file to remove the possible  duplication

File: ui/components/connections/index.js (hunk 1 above)
Comment (sudhanshutech) [reply ui/components/connections/index.js:614]: also is the sort sustained when move to next pages across al changed columns?

**Comment by saurabh100ni** [comment]
Let's discuss this on the Meshery dev call tomorrow,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Review by aabidsofi19 — DISMISSED** [review]
looks great

**Review by sudhanshutech — APPROVED** [review]

**Comment by senali-d** [comment]
@Yashsharma1911 

- [x] Change chip color position
- [x] Remove chip close icon from MeshSync datatable
- [x] Fix issue in secondary datatable open in MeshSync

**Comment by leecalcote** [comment]
Just look at @senali-d go!

**Comment by Yashsharma1911** [comment]
> @Yashsharma1911
> 
> * [x]  Change chip color position
> * [x]  Remove chip close icon from MeshSync datatable
> * [x]  Fix issue in secondary datatable open in MeshSync

Thankyou Senali I'm testing changes, docker desktop worked for me

**Review by Yashsharma1911 — APPROVED** [review]
Moving forward with this PR added some screenshot of UI

items for next PR

- [ ] Remove hardcoded status in connection table
- [ ] Display status that are supported by the connection, rather than showing all defined statuses for every connection.
- [ ] Define and use RTK queries
- [ ] Support column filter in meshsync table
   -  @sudhanshutech I tried to add column filter for meshsync table though there are some unexpected behavior was coming when I tried to add support for column filter for both tables there are some logic improvements that need to made please have a look

Screenshot 2023-10-26 at 1 55 59 AM
Screenshot 2023-10-26 at 1 55 41 AM

**Pull request merged**